packages/block-editor/src/components/floating-toolbar/index.native.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { View, TouchableWithoutFeedback, I18nManager } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { ToolbarButton, Toolbar } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+import NavigateUpSVG from './nav-up-icon';
+import Breadcrumb from '../block-list/breadcrumb.native';
+
+const FloatingToolbar = ( {
+	clientId,
+	parentId,
+	showFloatingToolbar,
+	onNavigateUp,
+} ) =>
+	showFloatingToolbar && (
+		<TouchableWithoutFeedback accessible={ false }>
+			<View style={ styles.floatingToolbar }>
+				{ !! parentId && (
+					<Toolbar passedStyle={ styles.toolbar }>
+						<ToolbarButton
+							title={ __( 'Navigate Up' ) }
+							onClick={ () => onNavigateUp( parentId ) }
+							icon={
+								<NavigateUpSVG isRTL={ I18nManager.isRTL } />
+							}
+						/>
+						<View style={ styles.pipe } />
+					</Toolbar>
+				) }
+				<Breadcrumb clientId={ clientId } />
+			</View>
+		</TouchableWithoutFeedback>
+	);
+
+export default compose( [
+	withSelect( ( select ) => {
+		const {
+			getSelectedBlockClientId,
+			getBlockHierarchyRootClientId,
+			getBlockRootClientId,
+			getBlockCount,
+		} = select( 'core/block-editor' );
+
+		const clientId = getSelectedBlockClientId();
+
+		const isSelected = !! clientId;
+		const rootBlockId = getBlockHierarchyRootClientId( clientId );
+		const parentId = getBlockRootClientId( clientId );
+		const hasRootInnerBlocks = !! getBlockCount( rootBlockId );
+
+		const showFloatingToolbar = isSelected && hasRootInnerBlocks;
+
+		return {
+			clientId,
+			showFloatingToolbar,
+			parentId,
+		};
```

Comment:
I think it would be better to add one check before all selectors since all of them depends on `selectedClientId`

Something like 
```js
if( !! selectedClientId ) {
  const rootBlockId = getBlockHierarchyRootClientId( selectedClientId );
  const parentId = getBlockRootClientId( selectedClientId );
  const hasRootInnerBlocks = !! getBlockCount( rootBlockId );
  return {
    selectedClientId,
    showFloatingToolbar,
    parentId,
  }
}
return { }
```

Reply:
how about that:

```jsx
		const selectedClientId = getSelectedBlockClientId();

		if( ! selectedClientId ) return;

		const rootBlockId = getBlockHierarchyRootClientId( selectedClientId );

		return {
			selectedClientId,
			showFloatingToolbar: !! getBlockCount( rootBlockId ),
			parentId: getBlockRootClientId( selectedClientId ),
		};
```

Reply:
WDYT ?

Reply:
LGTM :)